include/nbl/asset/filters/CFlattenRegionsStreamHashImageFilter.h

```diff
@@ -0,0 +1,220 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#ifndef __NBL_ASSET_C_FLATTEN_REGIONS_STREAM_HASH_IMAGE_FILTER_H_INCLUDED__
+#define __NBL_ASSET_C_FLATTEN_REGIONS_STREAM_HASH_IMAGE_FILTER_H_INCLUDED__
+
+#include <type_traits>
+
+#include "nbl/core/declarations.h"
+
+#include "nbl/asset/filters/CMatchedSizeInOutImageFilterCommon.h"
+#include "nbl/asset/filters/CFlattenRegionsImageFilter.h"
+#include "blake/c/blake3.h"
+
+namespace nbl
+{
+namespace asset
+{
+class CFlattenRegionsStreamHashImageFilter : public CMatchedSizeInOutImageFilterCommon
+{
+	public:
+		virtual ~CFlattenRegionsStreamHashImageFilter() {}
+
+		struct ScratchMemory
+		{
+			core::smart_refctd_ptr<asset::ICPUImage> flatten; // for flattening input regions & prefilling with 0-value texels not covered by regions 
+			core::smart_refctd_ptr<asset::ICPUBuffer> heap; // for storing hashes, single hash is obtained from given miplevel & layer, full hash for an image is a hash of this stack
+		};
+		
+		class CState : public IImageFilter::IState
+		{
+			public:
+				virtual ~CState() {}
+
+				using hash_t = std::array<size_t, 4u>;
+
+				const ICPUImage*					inImage = nullptr;
+				hash_t								outHash = { {0} };
```

Comment:
this needs to be a `core::blake3_hash_t`

Reply:
ok but remember this is created from `master` where it doesn't exist yet

Reply:
this thing doesn't exist on current `HEAD`, once you merge I can adjust

Reply:
you have it on your branch I believe

Reply:
time to merge `asset_converter3` to this branch then

Reply:
done